dsc/workflows/base/__init__.py

```diff
@@ -313,30 +308,25 @@ def item_submissions_iter(self) -> Iterator[ItemSubmission]:
 
         MUST NOT be overridden by workflow subclasses.
         """
-        for item_metadata in self.item_metadata_iter():
+        batch_metadata = {
+            self.get_item_identifier(item_metadata): item_metadata
+            for item_metadata in self.item_metadata_iter()
+        }
+        for item_submission_db in ItemSubmissionDB.get_batch_items(self.batch_id):
+            item_submission = ItemSubmission.from_db(item_submission_db)
             self.submission_summary["total"] += 1
-            item_identifier = self.get_item_identifier(item_metadata)
-            try:
-                item_submission = ItemSubmission.from_batch_id_and_item_identifier(
-                    batch_id=self.batch_id, item_identifier=item_identifier
-                )
-            except DoesNotExist:
-                logger.warning(
-                    "Record "
-                    f"{ITEM_SUBMISSION_LOG_STR.format(batch_id=self.batch_id,
-                                      item_identifier=item_identifier)}"
-                    "not found. Verify that it been reconciled."
-                )
-                self.submission_summary["skipped"] += 1
-                continue
-
-            logger.info(f"Preparing submission for item: {item_identifier}")
+            logger.info(
```

Comment:
It's minor, but I'd propose `DEBUG` logging here.  Overall, I think we should lean into `DEBUG` when inside a loop unless we have a very good reason not too.

Also, a much fuzzier heuristic might be `DEBUG` can help understand when something goes wrong.  If the item loaded successfully, awesome, maybe we don't need to know that in the logs.